tox.ini

```diff
@@ -0,0 +1,73 @@
+[tox]
+envlist = bandit,black,isort,flake8,mypy,python3.9,python3.10
+
+[gh-actions]
+python =
+    3.9: python3.9
+    3.10: python3.10
+
+[testenv]
+deps =
+    -rrequirements.txt
+    -rrequirements-dev.txt
+whitelist_externals =
+    make
+    mkdir
+    rm
+passenv = TOX_ENV_DIR
+usedevelop = true
+
+commands =
+    pytest \
+        --cov-config .coveragerc --cov=pyarn --cov-report term \
+        --cov-report xml --cov-report html
+
+[testenv:black]
+description = black checks [Mandatory]
+skip_install = true
+deps =
+    # Pin the version of black and click to avoid a newer version causing tox to fail
+    black==22.6.0
+    click==8.0.3
+commands =
+    black --check --diff pyarn tests
+
+[testenv:isort]
+skip_install = true
+deps =
+    isort[colors]
+commands =
+    isort --check --diff --color pyarn tests
+
+[testenv:flake8]
+skip_install = true
+deps =
+    flake8==3.9.2
+    flake8-docstrings==1.6.0
+commands =
+    flake8
+
+[flake8]
+show-source = True
+max-line-length = 100
+exclude = venv,.git,.tox,.env,pyarn/parsetab.py,pyarn/parser.out
+ignore = D102,D107,D101,D103,D105,D400,E226,D300,D104
+per-file-ignores =
+    # Ignore missing docstrings in the tests, migrations and missing/multiple whitespace
+    tests/*:D101,D102,D103,E226,E241,E231
+    ./pyarn/parser.py: D400,D403,D205,D209,D208,D103
+;     /tests/test_lexer.py:
+
+
+[testenv:bandit]
+skip_install = true
+deps =
+    bandit
+commands =
+    bandit -r pyarn
+
+[testenv:mypy]
+skip_install = true
+commands =
+    mypy -p pyarn --install-types --non-interactive \
+    --exclude pyarn/parsetab.py --ignore-missing-imports
```

Comment:
Do we have to `--ignore-missing-imports`?

Reply:
we have to test, in OSBS adn cachito, not all deps/imports had definitions for mypy

Reply:
Yes, instead we get: 

> mypy run-test: commands[0] | mypy -p pyarn --install-types --non-interactive --exclude pyarn/parsetab.py
pyarn/lockfile.py:22: error: Skipping analyzing "ply": module is installed, but missing library stubs or py.typed marker  [import]
pyarn/lockfile.py:22: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 5 source files)

Reply:
IIRC mypy is able to do some basic checks with even without annotations. Because it found some bugs inside OSBS w/o annotations

Reply:
With `--check-untyped-defs` yes, without it I don't think mypy checks anything
```python
def obviously_wrong():
    a = 1 + "two"


def obviously_wrong2() -> None:
    a = 1 + "two"
```
```shell
$ mypy x.py 
x.py:6: error: Unsupported operand types for + ("int" and "str")
Found 1 error in 1 file (checked 1 source file)

$ mypy x.py --check-untyped-defs
x.py:2: error: Unsupported operand types for + ("int" and "str")
x.py:6: error: Unsupported operand types for + ("int" and "str")
Found 2 errors in 1 file (checked 1 source file)
```

Reply:
Since mypy doesn't seem to add anything, I'll remove it.